Improve MIME type detection for image URLs

Seayon opened the pull request:
Implement fallback to content-based detection using Pillow when URL-based guess fails, enhancing accuracy while maintaining efficiency.

# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

Fixes https://github.com/langgenius/dify/issues/6529

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- I uploaded an image to Aliyun OSS that was actually in JPEG format but had a PNG file extension. When I tried to use Anthropic's LLM for image recognition, it resulted in an error.
After making the modifications, I used the same image again, and the API call was successful. Everything worked normally.
- [ ] Test B

laipz8200 commented:
Hello @Seayon, sorry for replying so late, could you please sync your branch with the main branch first? I think this PR could be merged.

Commit:
Improve MIME type detection for image URLs

Implement fallback to content-based detection using Pillow when URL-based guess fails, enhancing accuracy while maintaining efficiency.

Seayon commented:
> Hello @Seayon, sorry for replying so late, could you please sync your branch with the main branch first? I think this PR could be merged.

Hello @laipz8200 

Thank you for your review. I've synced the branch with the main branch and performed some cleanup. The branch should now be up-to-date, and the commits are clean. I look forward to your approval for merging this PR.

Let me know if you need any further information or changes.

laipz8200 reviewed (approved):
Thanks!
